src/compiler/types.ts

```diff
@@ -1669,10 +1669,8 @@ namespace ts {
         typeParameters: TypeParameter[];           // Type parameters (undefined if non-generic)
         outerTypeParameters: TypeParameter[];      // Outer type parameters (undefined if none)
         localTypeParameters: TypeParameter[];      // Local type parameters (undefined if none)
-    }
-
-    export interface InterfaceTypeWithBaseTypes extends InterfaceType {
-        baseTypes: ObjectType[];
+        resolvedBaseConstructorType?: Type;        // Resolved base constructor type of class
+        resolvedBaseTypes: ObjectType[];           // Resolved base types

```

Comment:
Yes, they're still deferred. The pattern we use elsewhere for *single* deferred values is a resolvedXXX property guarded by a getXXX function that returns the value. In cases where we defer the computation of *multiple* properties we use an XXXWithYYY type and return the object itself.